src/box/box.cc

```diff
@@ -3729,6 +3729,24 @@ box_deadline_is_up(void)
 		return false;
 }
 
+/* Force deadline of main cord on SIGURG. */
+void
+box_deadline_timeout_cb(int signum)
+{
+	(void)signum;
+	if (cord_is_main())
+		fiber_set_current_deadline({0, 0});
```

Comment:
So we have a certain probability that a fiber will be cancelled by `SIGURG`? This is unacceptable. I guess, we need to block signals on all threads except the main one, but I'm not sure. Please investigate.

Also, AFAIU signals should be set in `main.cc` and reset on fork.

Reply:
I can't understand what you mean.
Anyway, I moved this logic to `main.cc` and started to block `SIGURG` on child processes.

Reply:
AFAIK a signal can be received by any thread of a process. AFAICS you just ignore a signal if it was received by a process other than main (e.g. iproto or WAL). Am I missing something?

Reply:
Oh, thanks, I thought all the threads receives the signal.
I found out that we can use `pthread_sigmask` to block `SIGURG` on all threads instead of main one. Then the signal will be received by main thread only.

Reply:
Fixed it.